tests/test_deserialization.py

```diff
@@ -1178,6 +1178,18 @@ class AliasingUserSerializer(Schema):
         assert result["email"] == "foo@bar.com"
         assert "years" not in result
 
+    def test_deserialize_dotted_path_with_data_key_param(self):
+        class MySchema(Schema):
+            foo = fields.Str(data_key="foo")
+            bar = fields.Str(data_key="data.bar")
+
+        sch = MySchema(unknown=EXCLUDE)
```

Comment:
Why do we need `EXCLUDE`? (Not sure why we need it in test above either.)

Reply:
If I don't use `EXCLUDE` the schema wants to have a `data`  field to load the top level json element "data".
In the real world case where I want to use this I'm happy with using `EXCLUDE` because the top level keys don't mean anything.

I think it would get complicated to support `unknown=RAISE` and schemas reading values from dotted paths, at least from my limited understanding!

Reply:
Oh, of course. I understand, now. Thanks.

(Answer to myself: in the test above, `EXCLUDE` was added to exclude `years` from the input.)